R/course.R

```diff
@@ -273,6 +295,25 @@ tidy_unzip <- function(zipfile, cleanup = FALSE) {
   invisible(target)
 }
 
+## takes a Dropbox or Google Drive URL as provided in the web browser, and
+## converts into an URL suitable for direct download (as expected by
+## tidy_download() for instance).
+convert_teaching_url <- function(url) {
+  stopifnot(is.character(url))
+  stopifnot(grepl("^http[s]?://", url))
+
+  if (grepl("drive.google.com", url)) {
+    id <- regmatches(url, regexpr("id=[^&]+&?", url))
+    return(glue::glue("https://drive.google.com/uc?export=download&{id}"))
+  }
+
+  if (grepl("dropbox.com/sh", url)) {
+    return(gsub("dl=0", "dl=1", url))
+  }
+
```

Comment:
I think we should probably use `ui_info()` here to say something like: "Not a recognized URL format. No changes made."